test/cgroups.bats

```diff
@@ -25,6 +25,37 @@ function teardown() {
 	[[ "$output" == "1234" ]]
 }
 
+@test "negative pids limit" {
+	if ! grep -qEw ^pids /proc/cgroups; then
+		skip "pids cgroup controller is not available"
+	fi
+	CONTAINER_PIDS_LIMIT=-1 start_crio
+
+	jq '	  .command'='["/bin/sleep", "600"]' \
+		"$TESTDATA"/container_config.json > "$TESTDIR"/container_pids_limit.json
+
+	ctr_id=$(crictl run "$TESTDIR"/container_pids_limit.json "$TESTDATA"/sandbox_config.json)
+
+	output=$(crictl exec --sync "$ctr_id" sh -c 'cat /sys/fs/cgroup/pids/pids.max 2>/dev/null || cat /sys/fs/cgroup/pids.max')
+	[[ "$output" == "max" ]]
+}
+
+@test "zero pids limit" {
+	if ! grep -qEw ^pids /proc/cgroups; then
+		skip "pids cgroup controller is not available"
+	fi
+	CONTAINER_PIDS_LIMIT=0 start_crio
+
+	jq '	  .command'='["/bin/sleep", "600"]' \
+		"$TESTDATA"/container_config.json > "$TESTDIR"/container_pids_limit.json
+
+	ctr_id=$(crictl run "$TESTDIR"/container_pids_limit.json "$TESTDATA"/sandbox_config.json)
+
+	output=$(crictl exec --sync "$ctr_id" sh -c 'cat /sys/fs/cgroup/pids/pids.max 2>/dev/null || cat /sys/fs/cgroup/pids.max')
+	# TODO where does this number come from?
+	[[ "$output" == "37725" ]]
```

Comment:
lots of duplicate code here... I put it up because I was surprised by this value on my machine

Reply:
any theories why this value is what it is? @kolyshkin @giuseppe @mrunalp

Reply:
This comes from a parent cgroup I guess, and is set by systemd. For me, it's 18930. I haven't checked but believe systemd sets it depending on available RAM. Which probably means your laptop has 32 GB RAM, as mine has humble 16 :smile: 


```console
[kir@kir-rhat cri-o]$ cat /proc/self/cgroup 
0::/user.slice/user-1000.slice/user@1000.service/vte-spawn-00c4c013-d8a2-4bd4-997e-c42605190449.scope
[kir@kir-rhat cri-o]$ cat /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/vte-spawn-00c4c013-d8a2-4bd4-997e-c42605190449.scope/pids.max 
18930
```